src/main/java/com/laserfiche/repository/api/clients/impl/SearchesClientImpl.java

```diff
@@ -123,11 +134,30 @@ private CompletableFuture<ODataValueContextOfIListOfContextHit> doGetSearchConte
 
     @Override
     public CompletableFuture<ODataValueContextOfIListOfContextHit> getSearchContextHitsNextLink(String nextLink,
-            int maxPageSize) {
+            Integer maxPageSize) {
         return doGetSearchContextHits(nextLink, null, null, null, mergeMaxSizeIntoPrefer(maxPageSize, null), null, null,
                 null, null, null);
     }
 
+    @Override
+    public CompletableFuture<Void> getSearchContextHitsForEach(
+            Function<CompletableFuture<ODataValueContextOfIListOfContextHit>, CompletableFuture<Boolean>> callback,
+            Integer maxPageSize, String repoId, String searchToken, Integer rowNumber, String prefer, String select,
+            String orderby, Integer top, Integer skip, Boolean count) throws InterruptedException, ExecutionException {
+        prefer = mergeMaxSizeIntoPrefer(maxPageSize, prefer);
+        CompletableFuture<ODataValueContextOfIListOfContextHit> response = getSearchContextHits(repoId, searchToken,
+                rowNumber, prefer, select, orderby, top, skip, count);
+        while (response != null && callback
+                .apply(response)
+                .get()) {
```

Comment:
If we use `join()`, would it save us those: `throws InterruptedException, ExecutionException`?

Reply:
Reference: https://stackoverflow.com/questions/45490316/completablefuturet-class-join-vs-get

Reply:
both methods potentially throw exceptions, but join's exceptions are unchecked, while get's exceptions are checked. http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/concurrent/CompletableFuture.java
So, using join, we don't need the throws declaration. However I am not sure which one is better.
Currently, when a client calls this method in their code, as the method throws checked exceptions, the client is alerted by the compiler and need to make their code safe against these exceptions, if they are thrown. So, while this might be a burden for them, it helps them have more robust code. 
If we use join, then it might happen that a client who uses this method is not aware that what kinds of exceptions might occur, and do not use any exception handling, reducing their code's robustness. 
I'm ok with both but are personally prefer throwing checked exceptions. It's good to mob with Paolo about this.